spec/unit/type/k5login_spec.rb

```diff
@@ -131,6 +152,46 @@ def resource(attrs = {})
           end
         end
       end
+
+      context "#stat" do
+        let(:file) { described_class.new(:path => path) }
+        it "should return nil if the file does not exist" do
+          file[:path] = make_absolute('/foo/bar/baz/non-existent')
+
+          expect(file.stat).to be_nil
+        end
+
+        it "should return nil if the file cannot be stat'ed" do
+          dir = tmpfile('link_test_dir')
+          child = File.join(dir, 'some_file')
+          Dir.mkdir(dir)
+          File.chmod(0, dir)
+
+          file[:path] = child
+
+          expect(file.stat).to be_nil
```

Comment:
You haven't created the file for this test. If the user running the test is root, then we'd be able to read through the directory and you'd get nil because the file doesn't exist, not because you can't stat it.

I guess it kind of protects the test, if you happen to run the specs as root, you get nil because the file doesn't exist, and if your a user you get nil because you can't access the through the directory.